### PR TITLE
nodejs: instance based auth for bounds actions

### PR DESCRIPTION
**Pull request by schiwekM**
Hi colleagues,

this is for 8.3 feature of supporting instance based restrictions on bound actions and functions.

BR,
Marten

### CODE REVIEW

**Review by vkozyura — COMMENTED** [review]
@schiwekM: We should also add info that the restrict annotations only works on entity level, but not on action level

**Comment by schiwekM** [comment]
Should be done now - thanks for mentioning.

**Review by vkozyura — APPROVED** [review]

File: guides/security/authorization.md [hunk 1]
```diff
@@ -341,10 +341,11 @@ Restrictions can be defined on different types of CDS resources, but there are s
 | CDS Resource    | `grant` | `to` |      `where`      | Remark        |
 |-----------------|:-------:|:----:|:-----------------:|---------------|
 | service         |  <Na/>  | <Y/> |       <Na/>       | = `@requires` |
-| entity          |  <Y/>   | <Y/> |       <Y/>        |               |
-| action/function |  <Na/>  | <Y/> | <Na/><sup>1</sup> | = `@requires` |
+| entity          |  <Y/>   | <Y/> | <Y/><sup>1</sup>  |               |
+| action/function |  <Na/>  | <Y/> | <Na/><sup>2</sup> | = `@requires` |
 
-> <sup>1</sup> Node.js supports static expressions *that don't have any reference to the model* such as `where: $user.level = 2`. <br>
+> <sup>1</sup>For bound actions and functions, which are not bound against collection, Node.js supports Instance-Based Authorization on entity level - `where` clauses *containing references to the model* such as `where: CreatedBy = $user`. For all bound actions and functions Node.js supports simple static expressions *that don't have any reference to the model* on entity level such as `where: $user.level = 2`.
```
Comment (renejeglinsky) [inline guides/security/authorization.md:347]:
```suggestion
> <sup>1</sup>For bound actions and functions that aren't bound against a collection, Node.js supports instance-based authorization at the entity level. For example, you can use `where` clauses that *contain references to the model*, such as `where: CreatedBy = $user`. For all bound actions and functions, Node.js supports simple static expressions at the entity level that *don't have any reference to the model*, such as `where: $user.level = 2`.
```

File: guides/security/authorization.md [hunk 2]
```diff
@@ -341,10 +341,11 @@ Restrictions can be defined on different types of CDS resources, but there are s
...
+> <sup>2</sup> For unbound actions and functions Node.js supports simple static expressions *that don't have any reference to the model* such as `where: $user.level = 2`.
```
Comment (renejeglinsky) [inline guides/security/authorization.md:348]:
```suggestion
> <sup>2</sup> For unbound actions and functions, Node.js supports simple static expressions that *don't have any reference to the model*, such as `where: $user.level = 2`. 
```

File: guides/security/authorization.md [hunk 3]
```diff
@@ -494,7 +495,7 @@ The condition defined in the `where`-clause typically associates domain data wit
 - `UPDATE` (as reject condition)
 - `DELETE` (as reject condition)
 
- > <sup>1</sup> Node.js supports _static expressions_ *that don't have any reference to the model* such as `where: $user.level = 2` for all events including action and functions.
+ > <sup>1</sup> Node.js supports _static expressions_ *that don't have any reference to the model* such as `where: $user.level = 2` for all events.
```
Comment (renejeglinsky) [inline guides/security/authorization.md:498]:
```suggestion
 > <sup>1</sup> Node.js supports _static expressions_ that *don't have any reference to the model* such as `where: $user.level = 2` for all events.
```

**Review by renejeglinsky — APPROVED** [review]

**Pull request merged**